docs/core/building/index.md

```diff
@@ -0,0 +1,107 @@
+---
+title: Build .NET Core from source
+description: Build .NET Core from source
+keywords: .NET, .NET Core, source, build
+author: beleroy
+ms.author: mairaw
+ms.date: 03/29/2017
+ms.topic: article
+ms.prod: .net-core
+ms.devlang: dotnet
+ms.assetid: 8b49079c-6ede-429a-92d7-ecd2fda1ab0e
+---
+
+# Build .NET Core from source
+
+The ability for anyone to build .NET Core from its source code is important in multiple ways: it makes it easier to port .NET Core to new platforms, it enables contributions and fixes to the product, and it enables the creation of custom versions of .NET.
+This section of the documentation gives guidance to developers who want to build and distribute their own versions of .NET Core.
+
+## How to build from source
+
+### Building the CLR
+
+The source code for the .NET Core CLR can be found in [the `dotnet/coreclr` repository on GitHub](https://github.com/dotnet/coreclr/).
+
+The build currently depends on Git, CMake, Python and a C++ compiler.
+Once these prerequisites are installed, the CLR is built by invoking the 'build' script (`build.cmd` or `build.sh`) at the base of [the Core CLR repository](https://github.com/dotnet/coreclr/).
+
+The details of installing the components differ depending on the operating system.  See the following pages based on your OS.
+There is no cross-building across OS (only for ARM, which is built on X64).  
+You have to be on the particular platform to build that platform.  
+
+ * [Windows Build Instructions](https://github.com/dotnet/coreclr/blob/master/Documentation/building/windows-instructions.md)
+ * [Linux Build Instructions](https://github.com/dotnet/coreclr/blob/master/Documentation/building/linux-instructions.md)
+ * [macOS Build Instructions](https://github.com/dotnet/coreclr/blob/master/Documentation/building/osx-instructions.md)
+ * [FreeBSD Build Instructions](https://github.com/dotnet/coreclr/blob/master/Documentation/building/freebsd-instructions.md) 
+ * [NetBSD Build Instructions](https://github.com/dotnet/coreclr/blob/master/Documentation/building/netbsd-instructions.md)
+
+The build has two main 'buildTypes':
+
+ * Debug (default)- This compiles the runtime with additional runtime checks (asserts). These checks slow runtime execution but are really valuable for debugging, and is recommended for normal development and testing.  
+ * Release - This compiles without any development time runtime checks.  This is what end users will use but can be difficult to debug. Passing 'release' to the build script select this.
```

Comment:
inline code instead of quotes?

This compiles without any development time runtime checks -> compiles the runtime without any runtime checks.

This is what end users will use but can be difficult to debug. -> It can be difficult to debug. Recommended for deployment on production environments. ?

Passing 'release' to the build script select this. -> Pass `release` to the build script to select this type.